lightgbm/src/test/python/synapseml/lightgbm/test_lightgbm.py

```diff
@@ -0,0 +1,122 @@
+import os
+import shutil
+import json
+import unittest
+from pyspark.sql import SparkSession, Row
+from pyspark.ml import PipelineModel, Transformer
+from pyspark.ml.util import DefaultParamsReadable, DefaultParamsWritable
+from pyspark.ml.linalg import Vectors, VectorUDT
+from pyspark.sql.types import ArrayType, IntegerType
+import pyspark.sql.functions as f
+from synapse.ml.stages import SelectColumns
+import synapse.ml.lightgbm as lgbm
+
+class StringArrayToVectorTransformer(Transformer, DefaultParamsReadable, DefaultParamsWritable):
```

Comment:
is this a throwaway piece of code to just test serialization of a custom model?

Reply:
I was not exactly sure on the interpretation of this but stripped a fair amount of logic from the test and changed the function name/description to reflect better what it is for. Based on the update, let me know if you think if it should be further reduced. I have the single vs pipeline serialization test to demonstrate the standard single-model Python serialization/deserialization was not affected along with the pipeline test demonstrating the mentioned issue is resolved.